### ISSUE TITLE
Allowing incremental container copies

### ISSUE THREAD

**Issue by rkelleyrtp** [lxc/lxd#3605]
# Required information

 * Distribution: Ubuntu 
 * Distribution version: 16.04
 * The output of "lxc info" or if that fails:
   * Kernel version:4.4.0-83-generic
   * LXC version:  2.15
   * LXD version: 2.15
   * Storage backend in use: btrfs

# Issue description
Enhancement to allow incremental container copies - either on the same host or remote host.

Currently, doing an "lxc copy" produces an initial copy of the container, and running multiple copies of the same container gives an error message.  Since container live migration is hit-or-miss at this time, we need to ability to run multiple "lxc copy" commands to get the source and destination in sync as much as possible before starting the remote container.

Today, we accomplish this via "lxc copy" and rsync.  Like this:
* lxc copy server-01:container-01   server-02:container-02
* rsync -Warvx --delete-after server-01:container-01/rootfs server-02:container02
* lxc stop server-01:container-01
* rsync -Warvx --delete-after server-01:container-01/rootfs server-02:container02
* lxc start server-02:container-02

# A brief description of what failed or what could be improved.
Ideally, we could issue a command like "lxc copy --update server-01:container-01   server-02:container-02".  This would simply update the remote with the differences (either via zfs, btrfs, or standard rsync).  If the remote does not exist, simply perform an initial copy.  If the remote already exists, prompt the user to allow the update then sync the local to remote.  Optionally enable the "--force" to force a sync between the two containers without prompting the user.

**Comment by stgraber** [comment]
Duplicate of #3326
